### ISSUE TITLE
Additional evaluation for the pipeline

### ISSUE THREAD

**Issue by gsarti** [jinensetpal/evalita#1]
Hey @jinensetpal ,

Raising an issue instead of sending an email since it is more practical for me to keep track of things inside Github. I was looking to your evaluation script and what can be probably done to make the evaluation of your approach more sound is the following:

- Shuffle the dataframe before taking the X and y; we may never know if entries are ordered for some reason, and this would greatly affect the performances of the model (probably not, but still worth doing as a good practice).

- I saw that you were loading some pickled test sets for evaluation: I guess those are a portion of the original training that you set apart for testing. Would you mind doing a 5-fold cross-validation instead? This would ensure that a) there's no data leakage between train and test and b) that current results aren't due to a "lucky split".

Best,

Gabriele

**Comment by jinensetpal** [comment]
Understood - will implement changes, thank you!

**Comment by jinensetpal** [comment]
Changes made - 
Accuracy: 80.87% (+/- 2.26%)

**Comment by gsarti** [comment]
Looks cool @jinensetpal , two suggestions for you:

- sklearn.metrics.classification_report does exactly what you want to do, but better since it allows for micro-macro averaging. Try it for evaluation!

- In all folds, we can clearly see that recall is almost 100%, while precision lags behind at 70%. This concretely means that most memes are correctly classified as such, but many non-memes are classified as memes. Do you have some ideas on how to improve this aspect? It can involve, for example, lowering the rejection bar to see how the precision-recall proportions vary.

Let me know!

**Comment by jinensetpal** [comment]
Noted - will work on this!

**Comment by jinensetpal** [comment]
system built - closing issue!
